build-system/compile/single-pass.js

```diff
@@ -559,57 +562,63 @@ exports.singlePassCompile = async function(entryModule, options) {
     .then(wrapMainBinaries)
     .then(intermediateBundleConcat)
     .then(eliminateIntermediateBundles)
-    .then(thirdPartyConcat)
-    .then(removeInvalidSourcemaps)
-    .catch(err => {
-      err.showStack = false; // Useless node_modules stack
-      return Promise.reject(err);
-    });
+    .then(thirdPartyConcat);
```

Comment:
Can the steps after `compile` be run directly on the vinyl-fs files without having to write / rewrite? I was thinking you could consolidate `compile()` and `singlePassCompile()`, and inject these steps before the `gulp.dest('.')` at the end.

Reply:
Yes, but I think that's best as a follow up PR.

Reply:
SG. Maybe add a TODO?